server/routerlicious/packages/lambdas/src/alfred/index.ts

```diff
@@ -1100,12 +1122,35 @@ export function configureWebSocketServices(
 							: [contentBatch];
 
 						for (const content of contents) {
-							const signalMessage: ISignalMessage = {
-								clientId,
-								content,
-							};
+							let signalMessage: ISignalMessage;
+							let roomId: string;
+							const feature_submit_signals_v2 = "submit_signals_v2";
+
+							if (
```

Comment:
A short comment before the conditional might be nice
```suggestion
							// Decide whether to emit the signal to a single target client, or broadcast to everyone in the room
							if (
```